File: old/oldie/c#/linkedList/LinkedList.cs
using System;
using  System.Collections.Generic;

namespace linkedList
{
    class LinkedList
    {
        public Node root;
        public List<int> llst = new List<int>();

        // Constructor
        public LinkedList()
        {
            root = null;
        }


        // Insert Start
        public void insertStart(int value)
        {
            Node a = new Node(value);
            if(root == null)
            {
                root = a;
            }
            else
            {
                Node b = root;
                root = a;
                root.next = b;
            }
        }

        // Insert End
        public void insertEnd(int value)
        {
            Node a = new Node(value);
            if (root == null)
            {
                root = a;
            }
            else
            {
                ise(root, a);
            }
        }

        public void ise(Node node, Node newnode)
        {
            if (node.next == null)
            {
                node.next = newnode;
            }
            else
            {
                ise(node.next, newnode);
            }
        }
        
        // Remove Start
        public void removeStart()
        {
            if (root == null)
            {
                Console.WriteLine("Linked List is empty");
            }
            else if (root.next == null)
            {
                root = null;
            }
            else
            {
                Node n = root.next;
                root = n;
            }
        }

        // Remove End
        public void removeEnd()
        {
            if (root == null)
            {
                Console.WriteLine("Linked List is empty");
            }
            else if (root.next == null)
            {
                root = null;
            }
            else
            {
                remove(root);
            }
        }

        public void remove(Node node)
        {
            if (node.next.next == null)
            {
                node.next = null;
            }
            else
            {
                remove(node.next);
            }
        }

        // Remove Value
        public void removeValue(int value)
        {
            if(root == null)
            {
                Console.WriteLine("Linked List is empty!");
            }
            else if (root.data == value && root.next == null)
            {
                root = null;
            }
            else if (root.data != value && root.next == null)
            {
                Console.WriteLine("Linked List does not contain value!");
            }
            else if (root.data == value && root.next != null)
            {
                Node temp = root.next;
                root = temp;
            }
            else
            {
                rm(root, value);
            }
        }

        public void rm(Node node, int value)
        {
            if (node.next.data == value)
            {
                if (node.next.next == null)
                {
                    node.next = null;
                }
                else
                {
                    Node temp = node.next.next;
                    node.next = temp;
                }
            }
            else
            {
                if (node.next.next != null)
                {
                    rm(node.next, value);
                }
                else
                {
                    Console.WriteLine("Linked list does not contain value!");
                }
            }
        }

        // Traverse
        public void traverse()
        {
            if(root == null)
            {
                Console.WriteLine("Linked List is empty!");
            }
            else
            {
                llst.Add(root.data);

                if(root.next != null)
                {
                    tra(root.next);
                }
            }
            foreach (int item in llst)
            {
                Console.WriteLine(item);
            }
        }

        public void tra(Node node)
        {
            llst.Add(node.data);
            if (node.next != null)
            {
                tra(node.next);
            }
        }
    }
}

File: old/oldie/py/new/linked-lists/linkedlist.py
#!/usr/bin/python3
# Author:   @AgbaD || @agba_dr3


# ================ Linked List =================== #

class Node:

    def __init__(self, value=None):
        self.value = value
        self.next = None


class LinkedList:
    def __init__(self):
        self.root = None

    def insert_start(self, value):
        if not self.root:
            self.root = Node(value)
        else:
            new_node = Node(value)
            cur = self.root
            self.root = new_node
            new_node.next = cur

    def insert_end(self, value):
        if not self.root:
            self.root = Node(value)
        else:
            self.insert(self.root, value)

    def insert(self, node, value):
        if node.next:
            self.insert(node.next, value)
        else:
            new = Node(value)
            node.next = new

    def remove_start(self):
        if not self.root:
            print("Linked list has no values")
            return
        else:
            if self.root.next:
                temp = self.root.next
                self.root = temp
            else:
                self.root = None

    def remove_end(self):
        if not self.root:
            print("Linked list has no values")
            return
        elif self.root.next:
            self.remove(self.root)
        else:
            self.root = None

    def remove(self, node):
        if node.next:
            if node.next.next:
                self.remove(node.next)
            else:
                node.next = None

    def remove_value(self, value):
        if not self.root:
            print("Linked list has no values")
            return
        elif self.root.value == value and not self.root.next:
            self.root = None
        elif self.root.value != value and not self.root.next:
            print("Linked list does not contain value")
            return
        elif self.root.value == value and self.root.next:
            temp = self.root.next
            self.root = temp
        else:
            self.rm(self.root, value)

    def rm(self, node, value):
        if node.next.value == value:
            if node.next.next:
                temp = node.next.next
                node.next = temp
            else:
                node.next = None
        else:
            if node.next.next:
                self.rm(node.next, value)
            else:
                print("Linked list does not contain value")
                return

    def traverse(self):
        if not self.root:
            print("Linked list has no values")
            return
        lst = []
        lst.append(self.root.value)
        if self.root.next:
            self.add(lst, self.root)
        print(lst)

    def add(self, lst, node):
        lst.append(node.next.value)
        if node.next.next:
            self.add(lst, node.next)


if __name__ == "__main__":
    l = LinkedList()
    l.traverse()

    l.insert_start(7)
    l.insert_start(5)
    l.insert_start(3)

    l.traverse()

    l.insert_end(9)
    l.insert_end(11)

    l.traverse()

    l.remove_end()

    l.traverse()

    l.remove_start()

    l.traverse()

    l.remove_value(7)

    l.traverse()